#10: List of stored links

mdrlzy opened the pull request:
#10

mdrlzy committed:
orbit-mvi added

mdrlzy committed:
List of stored links implemented

kirillt commented on app/build.gradle:
```diff
@@ -50,33 +50,52 @@ android {
 dependencies {
+    def arklibVersion = '0.1.0-SNAPSHOT-f81e34a657'
+    releaseImplementation "space.taran:arklib:$arklibVersion"
+//    debugImplementation "space.taran:arklib-debug:$arklibVersion"
```
//

mdrlzy replied:
Actually debug version is very slow and doesn't seem to be needed

kirillt commented on app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt:
```diff
@@ -1,61 +1,93 @@
+            downloadPreview = false
+        )
+        link.imagePath?.let {
+            it.copyTo(linkPreviewPath(root, link.url))
```
Later, this image should be handled via Previews API of `arklib`.

mdrlzy replied:
Sure

kirillt commented on app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt:
```diff
@@ -1,61 +1,93 @@
+        val parseLinksJob = linkFolder
+            .listChildren()
+            .filter {
+                !it.isDirectory() && it.extension == "link"
```
Ideally, we should setup Metadata processor and receive list of resources from it, filtering them by kind.

mdrlzy replied:
Now we are saving wrong meta for links

kirillt commented on app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt:
```diff
@@ -1,61 +1,93 @@
 
-private data class JsonLink(val url: String, val title: String, val desc: String)
\ No newline at end of file
+    private fun linkPreviewPath(root: Path, url: String) = root
```
Also should be handled by Previews API later.

mdrlzy replied:
Sure

kirillt commented on app/src/main/java/space/taran/arkshelf/data/LinkRepoImpl.kt:
```diff
@@ -1,14 +1,17 @@
+import javax.inject.Inject
 
-class LinkRepoImpl(
+class LinkRepoImpl @Inject constructor(
```
I'm curious, why do we need `NetworkStatus` here?

mdrlzy replied:
To determine that we got error precisely because of the lack of internet and not because server is unavailable for example

kirillt commented on app/src/main/java/space/taran/arkshelf/data/LinkRepoImpl.kt:
```diff
@@ -26,4 +29,6 @@ class LinkRepoImpl(
         withContext(Dispatchers.IO) {
```
Ideally, the function should be declared as `generateResource(resource: Link, rootPath: Path)` if we want to highlight the concepts from `arklib`.

mdrlzy replied:
Sure

kirillt commented on app/src/main/java/space/taran/arkshelf/presentation/searchedit/LinkListAdapter.kt:
```diff
@@ -0,0 +1,178 @@
+            .into(binding.ivPreview)
+
+        Glide
+            .with(binding.ivThumbnail)
```
Should it be visible in the list? I see only preview when I create new link.

mdrlzy replied:
Yes, thumbnail used in list in collapsed state.

kirillt reviewed (approved):
It's not final version of course, but that's a huge work so let's merge it and create new tasks.

mdrlzy committed:
Adopted arklib, LinkListAdapter, minor fixes

Bump arkfilepicker, compileSdk
Toast on link copy, async save if url external
APK ABI filter
fix folder change, disable dark theme
